### PR TITLE
G-API Documentation: first submission

### PR DESCRIPTION
**Pull request by dmatveev**
This PR introduces a number of new OpenCV documentation chapters for
Graph API module.

In particular, the following topics are covered:
- Introduction & background information;
- High-level design overview;
- Kernel API;
- Pipeline example.

All changes are done in Markdown files, no headers, etc modified.

Doxygen references for main API classes will be added later.

Also, a tutorial will be introduced soon (in the common Tutorials place)

```
**WIP**
force_builders=Docs
```

### CODE REVIEW

File: modules/gapi/doc/00-root.markdown [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+input/output data references (in this example, `in` and `out`
+cv::GMat objects, respectively) as parameters and reconstructs the
+call graph based on all the data flow between `in` and `out`.
+
+cv::GComputation is a thin object in sense that it just captures which
+operations form up a computation. However, it can be used to execute
+computations -- in the following processing loop, every captured frame (a
+cv::Mat `input_frame`) is passed to cv::GComputation::apply().
+
+![Example pipeline running on sample video 'vtest.avi'](pics/demo.png)
```
Comment (alalek) [inline modules/gapi/doc/00-root.markdown]: Please use .jpg format (with ~90% quality) for real-like images.
PNG is not very good for this purpose.

Other images (schemes) are fine to store in png format.

File: modules/gapi/doc/00-root.markdown [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      Image processing functions: color space conversions, various
+      filters, etc.
+
+# API Example {#gapi_example}
+
+A very basic example of G-API pipeline is shown below:
+
+@code{.cpp}
+    #include <opencv2/opencv.hpp>
+    #include <opencv2/gapi.hpp>
```
Comment (alalek) [inline modules/gapi/doc/00-root.markdown]: It is better to extract tutorials code into separate files:
- create gapi folder here: https://github.com/opencv/opencv/tree/4.0.0-beta/samples/cpp/tutorial_code (or here: `modules/gapi/docs/code` - for better compatibility with standalone gapi)
- this code is checked at least for compilation
- use `@example` or `@snippet` to reference from documentation part.

Comment (dmatveev) [reply modules/gapi/doc/00-root.markdown]: This is not a tutorial, this is an [almost useless] API sample -- just for an introduction.

I will add a more detailed tutorial into a proper location as it gets ready.

Comment (dmatveev) [reply modules/gapi/doc/00-root.markdown]: Ah got it, will export the code into a distinct location, thanks!

File: modules/gapi/doc/00-root.markdown (hunk 1 above)
Comment (dmatveev) [reply modules/gapi/doc/00-root.markdown]: Okay

File: modules/gapi/doc/00-root.markdown [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+      Image processing functions: color space conversions, various
+      filters, etc.
+
+# API Example {#gapi_example}
+
+A very basic example of G-API pipeline is shown below:
+
+@code{.cpp}
+    #include <opencv2/opencv.hpp>
```
Comment (alalek) [inline modules/gapi/doc/00-root.markdown]: `opencv.hpp` is deprecated header.
We don't use it in OpenCV samples.

```
#include <opencv2/core.hpp>
#include <opencv2/videoio.hpp>
#include <opencv2/highgui.hpp>
```

File: modules/gapi/doc/00-root.markdown [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        cv::VideoCapture cap;
+        if (argc > 1) cap.open(argv[1]);
+        else cap.open(0);
+        CV_Assert(cap.isOpened());
+
+        cv::GMat in;
+        cv::GMat vga      = cv::gapi::resize(in, cv::Size(), 0.5, 0.5);
+        cv::GMat gray     = cv::gapi::BGR2Gray(vga);
+        cv::GMat blurred  = cv::gapi::blur(gray, cv::Size(5,5));
+        cv::GMat edges    = cv::gapi::canny(blurred, 32, 128, 3);
```
Comment (alalek) [inline modules/gapi/doc/00-root.markdown]: canny -> Canny

to fix compilation

**Comment by dmatveev** [comment]
Ok, will cherry-pick your CMake stuff here and move more sources to files

File: modules/gapi/doc/00-root.markdown [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    - @subpage gapi_imgproc
+
+      Image processing functions: color space conversions, various
+      filters, etc.
+
+# API Example {#gapi_example}
+
+A very basic example of G-API pipeline is shown below:
+
+@include modules/gapi/doc/code/api_example.cpp
```
Comment (alalek) [inline modules/gapi/doc/00-root.markdown]: OpenCV Doxyfile contains `modules/gapi/doc` in `EXAMPLE_PATH` list, so using of `code/api_example.cpp` is enough to point on the file.

This is another reason to add "gapi_" prefix to file names (to keep them unique).

**Comment by dmatveev** [comment]
@alalek done - your commit picked, code exported, now compiles.

**Comment by alalek** [comment]
BTW, What is about of removal of "cv::" from the code (adding `using namespace cv;` after includes) ?

/cc @vpisarev

**Comment by dmatveev** [comment]
Personally I'd prefer using namespaces everywhere - remember we do have `own::Mat`, `own::Point` and all that stuff now.

**Comment by dmatveev** [comment]
Hey folks @alalek @vpisarev! Is anything else required for this patch to be merged?

**Comment by dmatveev** [comment]
@alalek cherry-picked from your branch, thanks!

**Comment by dmatveev** [comment]
Thanks!

**Pull request merged**